sale_stock_release_channel_partner_by_date_delivery/models/sale_order.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    release_channel_id = fields.Many2one(
+        domain=(
+            "carrier_id and "
+            "['|', ('carrier_ids', '=', False), ('carrier_ids', 'in', [carrier_id])] "
+            "or []"
```

Comment:
i think you can just write
```suggestion
            "[('carrier_ids', 'in', [False, carrier_id])]"
```

Reply:
I think I tried but it wasn't returning the expected result. I'll check again.

Reply:
Proposed solution doesn't work.
I change the computation of this domain returned to the view, not sure it is the best approach but I could not think of a best one :man_shrugging: